cmd/minishift/cmd/console.go

```diff
@@ -68,6 +69,13 @@ func displayConsoleInMachineReadable(hostIP string, url string) {
 }
 
 func getHostUrl(api *libmachine.Client) string {
+	vmStatus, err := cluster.GetHostStatus(api, constants.MachineName)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error getting Minishift status: %s", err.Error()))
+	}
+	if vmStatus != machineState.Running.String() {
+		atexit.ExitWithMessage(1, "Error: Minishift is not running")
+	}
```

Comment:
https://github.com/minishift/minishift/issues/1996#issuecomment-366648348 we do need to do a bit extra for this.